src/lib.rs

```diff
@@ -196,21 +208,82 @@ fn do_camera_movement(
     let delta_device_pixels = current_pos - last_pos.unwrap_or(current_pos);
 
     for (cam, mut transform, projection) in &mut query {
-        if !cam.enabled
-            || !cam
-                .grab_buttons
-                .iter()
-                .any(|btn| mouse_buttons.pressed(*btn) && !mouse_buttons.just_pressed(*btn))
-        {
+        if !cam.enabled {
             continue;
         }
 
         let proj_area_size = projection.area.size();
-        let world_units_per_device_pixel = proj_area_size / window_size;
+
+        let mouse_delta = if !cam
+            .grab_buttons
+            .iter()
+            .any(|btn| mouse_buttons.pressed(*btn) && !mouse_buttons.just_pressed(*btn))
+        {
+            None
+        } else {
+            Some(delta_device_pixels * proj_area_size / window_size)
+        };
+
+        let mut direction = Vec2::ZERO;
+
+        if cam
+            .move_keys
+            .left
+            .iter()
+            .any(|key| keyboard_buttons.pressed(*key))
+        {
+            direction.x += 1.;
+        }
+
+        if cam
+            .move_keys
+            .right
+            .iter()
+            .any(|key| keyboard_buttons.pressed(*key))
```

Comment:
I think perhaps you could also do

```rust
keyboard_buttons.any_pressed(cam.move_keys.right)
```

Not sure which reads better?

Reply:
`any_pressed` requires ownership of `KeyCode` which does not implement `copy`. I think if we can avoid cloning here we should.